src/qtics/experiment.py

```diff
@@ -0,0 +1,206 @@
+            setattr(self, name, inst)
+            log.info("Added instrument %s.", name)
+
+    def all_instruments(self, func_name, *args, **kwargs):
+        """Apply function to all instruments."""
+        for name in self.inst_names:
+            if hasattr(self, name):
+                inst = getattr(self, name)
+                func = getattr(inst, func_name)
+                _ = func(*args, **kwargs)
+
+    def append_data_group(
+        self, group_name: str, parent_name="", datasets=None, **attributes
+    ):
+        """Save data appending to hdf5 file."""
+        with h5py.File(self.data_file, "a") as file:
+            if parent_name != "":
+                parent_group = file.require_group(parent_name)
+                group = parent_group.require_group(group_name)
+            else:
+                group = file.require_group(group_name)
+
+            if datasets is not None:
+                for data_name, data in datasets.items():
+                    group.create_dataset(data_name, data=data)
+            if attributes:
+                group.attrs.update(attributes)
+
+    def get_datasets_dict(self, data_file=""):
+        """Load the datasets of an hdf5 file as dictionary."""
+
+        def _recurse(h5file, path):
+            data = {}
+            group = h5file[path]
+            for key in list(group):
+                if key != "config":
+                    item = group.get(key)
+                    if isinstance(item, h5py.Dataset):
+                        data[key] = np.asarray(group.get(key))
+                    elif isinstance(item, h5py.Group):
+                        data[key] = _recurse(h5file, path + key + "/")
+            return data
+
+        if not data_file:
+            data_file = self.data_file
+
+        with h5py.File(data_file, "r") as h5file:
+            return _recurse(h5file, "/")
+
+    def save_config(self):
+        """Save experiment's attributes and instruments defaults."""
+        config_attr = {
+            key: getattr(self, key)
+            for key in dir(self)
+            if isinstance(getattr(self, key), (int, float, str, bool))
+            and not key.startswith("_")
+        }
+        self.append_data_group("config", **config_attr)
+        for inst_name in self.inst_names:
+            if hasattr(self, inst_name):
+                inst = getattr(self, inst_name)
+                self.append_data_group(inst_name, parent_name="config", **inst.defaults)
+
+
+class MonitorExperiment(BaseExperiment):
+    """Base monitoring experiment class."""
+
+    sleep: float = 5
+
+    def watch(self, event: Event):
+        """Run the experiment continuously until event is set."""
+        self.all_instruments("connect")
+        log.info("Running monitor %s", self.name)
+        while True:
+            time.sleep(self.sleep)
+            self.main()
+            if event.is_set():
+                log.info("Trigger event set, %s shutting down.", self.name)
+                return
```

Comment:
```suggestion
        while True and not event.is_set():
            time.sleep(self.sleep)
            self.main()
        else:
            log.info("Trigger event set, %s shutting down.", self.name)
```
This removes a line, so it's an incredible improvement